app/[...slug]/page.tsx

```diff
@@ -0,0 +1,55 @@
+import { getStoryblokApi, ISbStoriesParams, StoryblokClient } from '@storyblok/react/rsc';
+import StoryblokStory from '@storyblok/react/story';
+import { resolveRelations } from '@/utilities/resolveRelations';
+
+const activeEnv = process.env.NODE_ENV || 'development';
+
+export default async function Page({ params }) {
+  const { data } = await getStoryData(params);
+
+  return (
+    <StoryblokStory story={data.story} />
+  );
+};
+
+// Make sure to not export the below functions otherwise there will be a typescript error
+// https://github.com/vercel/next.js/discussions/48724
+async function getStoryData(params) {
+  let slug: string = params.slug ? params.slug.join('/') : 'home';
+  let sbParams: ISbStoriesParams = {
+    version: activeEnv === 'development' ? 'draft' : 'published',
+    cv: Date.now(),
+    resolve_relations: resolveRelations,
+  };
+
+  const storyblokApi: StoryblokClient = getStoryblokApi();
+  const story = await storyblokApi?.get(`cdn/stories/${slug}`, sbParams);
+
+  return story;
+};
+
+type pathsType = {
+  slug: string[];
+}
+
+async function generateStaticParams() {
+  const storyblokApi: StoryblokClient = getStoryblokApi();
+  let sbParams: ISbStoriesParams = {
+    version: activeEnv === 'development' ? 'draft' : 'published',
+    cv: Date.now(),
+  };
+
+  const { data: { links } } = await storyblokApi?.get('cdn/links', sbParams);
+  let paths: pathsType[] = [];
+
+  Object.keys(links).forEach((linkKey) => {
+    if (links[linkKey].is_folder || links[linkKey].slug === 'home') {
+      return;
+    }
+    const slug: string = links[linkKey].slug;
+    let splittedSlug = slug.split('/');
+    paths.push({ slug: splittedSlug });
+  });
+
+  return paths;
+};
```

Comment:
This is the nextjs 13 app way `generateStaticParams` that replaces the old `getStaticPaths` in the page directory approach. This creates all the routes that we need by pulling in the full list of page slugs that exist on Storyblok.
https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#dynamic-paths-getstaticpaths
https://nextjs.org/docs/app/api-reference/functions/generate-static-params